app/javascript/app/components/input/input-component.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import debounce from 'lodash/debounce';
+import cx from 'classnames';
+import { themr } from 'react-css-themr';
+
+import styles from './input-styles.scss';
+
+class Input extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.value
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.autofocus) {
+      this.inputRef.focus();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.value !== this.props.value) {
+      this.setState({ value: nextProps.value });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { autofocus } = this.props;
+    if (autofocus && autofocus !== prevProps.autofocus) {
+      this.inputRef.focus();
+    }
+  }
+
+  handleChange = value => {
+    this.setState({ value });
+    this.debouncedChange();
+  };
+
+  debouncedChange = debounce(() => {
+    const { onChange } = this.props;
+    if (onChange) {
+      this.props.onChange(this.state.value);
+    }
+  }, 300);
+
+  render() {
+    const { value } = this.state;
+    const {
+      theme,
+      className,
+      handleKeyUp,
+      disabled,
+      inputType,
+      placeholder,
+      label,
+      unit,
+      onBlur
+    } = this.props;
+    const inputProps = {
+      ref: el => {
+        this.inputRef = el;
+      },
+      type: inputType === 'number' ? 'number' : 'text',
+      className: cx(styles.input, className, theme.input, {
+        [theme.textArea]: inputType === 'textarea'
+      }),
+      onChange: e => this.handleChange(e.target.value),
+      onBlur: () => onBlur(value),
+      value: disabled ? 'n/a' : value,
```

Comment:
`n/a` text is not being displayed when `disabled` is set to `true` (the placeholder is still there)